streams/src/main/java/org/apache/kafka/streams/kstream/internals/TupleForwarder.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.state.internals.CachedStateStore;
+
+
+class TupleForwarder<K, V> {
```

Comment:
Though I understand from @dguy that with customizable state store we may have different mechanism regarding the store type, this tuple forwarder implementation is a bit hacky to me.

I'm wondering, could we instead still force the `CachingKeyValueStore` and `CachingWindowedKeyValueStore` in the DSL wrapping the user provided `StateStoreSupplier.get()`, and check that it is in the type of `KeyValueStore` and `WindowStore`, so that users do not need to worry about manually set caching with the customizable store supplier but just set that through the config?